source/module_hamilt_pw/hamilt_pwdft/forces.cpp

```diff
@@ -665,6 +745,119 @@ void Forces<FPTYPE, Device>::cal_force_ew(const UnitCell& ucell,
         aux[rho_basis->ig_gge0] = std::complex<double>(0.0, 0.0);
     }
 
+    // sincos op for cal_force_ew
+    
+    std::vector<FPTYPE> it_facts_host(this->nat);
+    std::vector<FPTYPE> tau_flat(this->nat * 3);
+    
+    // iterate over by lookup table
+    for (int iat = 0; iat < this->nat; iat++) {
+        int it = ucell.iat2it[iat];  
+        int ia = ucell.iat2ia[iat];  
+        
+        double zv;
+        if (PARAM.inp.use_paw)
```

Comment:
I would suggset deleting PAW code, we don't support it now. In future, we will implement a new one if  needed.

Reply:
I see. The current implementation is consistent with the original code so that all features keep the same only with the improved performance.